src/Microsoft.Extensions.Logging/LoggerFactoryExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    public static class LoggerFactoryExtensions
+    {
+        /// <summary>
+        /// Adds a filter that applies to <paramref name="providerName"/> and <paramref name="categoryName"/>, allowing logs with the given
+        /// minimum <see cref="LogLevel"/> or higher.
+        /// </summary>
+        /// <param name="factory">The <see cref="LoggerFactory" /> to apply the filter to.</param>
+        /// <param name="providerName">The name of the provider.</param>
+        /// <param name="categoryName">The name of the logger category.</param>
+        /// <param name="minLevel">The minimum <see cref="LogLevel"/> that logs from
+        /// <paramref name="providerName"/> and <paramref name="categoryName"/> are allowed.</param>
+        public static LoggerFactory AddFilter(this LoggerFactory factory, string providerName, string categoryName, LogLevel minLevel)
```

Comment:
Talked in person, the perf is about optimizing the instance methods because they can do smarter stuff w/ data structures. But then there's no reason to have extension methods, they might as well all be instance.